File: spring-backend/src/main/java/com/scanmyfood/backend/models/FoodNutrient.java
package com.scanmyfood.backend.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class FoodNutrient {
    private String name;
    private Quantity quantity;
}

File: flutter-app/lib/models/food_nutrient.dart
import 'package:read_the_label/models/quantity.dart';

class FoodNutrient {
  String name;
  Quantity quantity;

  FoodNutrient({
    required this.name,
    required this.quantity,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'quantity': quantity.toJson(),
    };
  }

  factory FoodNutrient.fromJson(Map<String, dynamic> json) {
    return FoodNutrient(
      name: json['name'] ?? '',
      quantity:
          Quantity.fromJson(json['quantity'] ?? {'value': 0.0, 'unit': ''}),
    );
  }
}
